### ISSUE TITLE
linearmodels fails to import, throws error: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject

### ISSUE THREAD

**Issue by blucap** [bashtage/linearmodels#327]
I get this error when I try to import linearmodels: "numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject" 

I run Anaconda, python 3.8.8 on Windows 10. It is a fresh installation of Anaconda Python on a new machine.

I tried to solve this problem by using an older NumPy version (1.6.1) in a separate environment. Alas. 

This is the output that shows which versions of libraries I use, and note the real path is not `c:\users\`, but `c:\users\myusername\` 

    Requirement already satisfied: linearmodels in c:\users\anaconda3\lib\site-packages (4.21)
    Requirement already satisfied: pandas>=0.24 in c:\users\anaconda3\lib\site-packages (from linearmodels) (1.2.3)
    Requirement already satisfied: pyhdfe>=0.1 in  c:\users\anaconda3\lib\site-packages (from linearmodels) (0.1.0)
    Requirement already satisfied: scipy>=1.2 in c:\users\anaconda3\lib\site-packages (from linearmodels) (1.6.1)
    Requirement already satisfied: statsmodels>=0.11 in c:\users\anaconda3\lib\site-packages (from linearmodels) (0.12.2)
    Requirement already satisfied: mypy-extensions>=0.4 in c:\users\anaconda3\lib\site-packages (from linearmodels) (0.4.3)
    Requirement already satisfied: property-cached>=1.6.3 in c:\users\anaconda3\lib\site-packages (from linearmodels) (1.6.4)
    Requirement already satisfied: numpy>=1.16 in c:\users\anaconda3\lib\site-packages (from linearmodels) (1.19.2)
    Requirement already satisfied: patsy in c:\users\anaconda3\lib\site-packages (from linearmodels) (0.5.1)
    Requirement already satisfied: Cython>=0.29.21 in c:\users\anaconda3\lib\site-packages (from linearmodels) (0.29.22)
    Requirement already satisfied: python-dateutil>=2.7.3 in c:\users\anaconda3\lib\site-packages (from pandas>=0.24->linearmodels) (2.8.1)
    Requirement already satisfied: pytz>=2017.3 in c:\users\anaconda3\lib\site-packages (from pandas>=0.24->linearmodels) (2021.1)
    Requirement already satisfied: six>=1.5 in c:\users\anaconda3\lib\site-packages (from python-dateutil>=2.7.3->pandas>=0.24->linearmodels) (1.15.0)

This is what proceeds he error:

    from linearmodels import PanelOLS
    Traceback (most recent call last):
    
      File "<ipython-input-2-6de63ba6aa3f>", line 1, in <module>
        from linearmodels import PanelOLS
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\__init__.py", line 38, in <module>
        from .asset_pricing.model import (
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\asset_pricing\__init__.py", line 1, in <module>
        from .model import LinearFactorModel, LinearFactorModelGMM, TradedFactorModel
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\asset_pricing\model.py", line 13, in <module>
        from linearmodels.asset_pricing.covariance import (
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\asset_pricing\covariance.py", line 9, in <module>
        from linearmodels.iv.covariance import (
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\iv\__init__.py", line 1, in <module>
        from .absorbing import AbsorbingLS, Interaction  # flake8: noqa
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\iv\absorbing.py", line 54, in <module>
        from linearmodels.panel.utility import (
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\panel\__init__.py", line 1, in <module>
        from linearmodels.panel.model import (
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\panel\model.py", line 28, in <module>
        from linearmodels.panel.utility import (
      File "C:\Users\Anaconda3\lib\site-packages\linearmodels\panel\utility.py", line 24, in <module>
        from linearmodels.panel._utility import _drop_singletons
      File "linearmodels\panel\_utility.pyx", line 1, in init linearmodels.panel._utility
    
    ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject

**Comment by bashtage** [comment]
This suggests that the wheel was built with NumPy 1.20.x, which is a bit surprising.  What happens if you upgrade to NumPy 1.19.5 or 1.20.1?

You should be able to install with 

```
pip uninstall linearmodels
pip install linearmodels --no-binary linearmodels --no-build-isolation
```

If you don't have a C compiler available, it will install without some speedups.

**Comment by bashtage** [comment]
Which linear models are you trying to use?  I suspect if the latest doesn't work, then one of the slightly older ones will.

**Comment by blucap** [comment]
Trying to run version 4.21.  

After installing the CodeBlocks C compiler and running `pip install linearmodels --no-binary linearmodels --no-build-isolation`, I get this at the end of 197 lines with a font coloured in red, see below (See for all lines, pastebin: [https://pastebin.com/sHNyesRs](https://pastebin.com/sHNyesRs)). 

    ERROR: Failed building wheel for linearmodels
    Failed to build linearmodels
    ERROR: Could not build wheels for linearmodels which use PEP 517 and cannot be installed directly

Anaconda offers NumPy only until 1.19.2, see [link](https://anaconda.org/anaconda/numpy).

**Comment by bashtage** [comment]
Should be fixed in the 4.22 release which is now posted.  Thanks for reporting.
